src/Memcached.php

```diff
@@ -2,252 +2,241 @@
-     * By default the Memcached instances are destroyed at the end of the request. To create an instance that
-     * persists between requests, you may specify a unique ID for the instance. All instances created with the
-     * same ID will share the same connection.
-     * @see http://ca2.php.net/manual/en/memcached.construct.php
-     */
-    public function setPersistentId(string $persistentId): void
+    public function has($key): bool
     {
-        $this->persistentId = $persistentId;
+        $this->cache->get($key);
+        return $this->cache->getResultCode() === \Memcached::RES_SUCCESS;
     }
 
     /**
-     * @param array $options options for Memcached.
-     * @see http://ca2.php.net/manual/en/memcached.setoptions.php
+     * Returns underlying \Memcached instance
+     * @return \Memcached
      */
-    public function setOptions(array $options): void
+    public function getCache(): \Memcached
     {
-        $this->options = $options;
+        return $this->cache;
     }
 
     /**
-     * @param string $username memcached sasl username.
-     * @see http://php.net/manual/en/memcached.setsaslauthdata.php
+     * Inits Memcached instance
      */
-    public function setUsername(string $username): void
+    private function initCache(): void
     {
-        $this->username = $username;
+        $this->cache = new \Memcached($this->persistentId);
     }
 
     /**
-     * @param string $password memcached sasl password.
-     * @see http://php.net/manual/en/memcached.setsaslauthdata.php
+     * Converts TTL to expiration
+     * @param $ttl
+     * @return int
      */
-    public function setPassword(string $password): void
+    private function ttlToExpiration($ttl): int
     {
-        $this->password = $password;
-    }
+        $ttl = $this->normalizeTtl($ttl);
 
-    protected function getValue(string $key, $default = null)
-    {
-        $value = $this->cache->get($key);
-
-        if ($this->cache->getResultCode() === \Memcached::RES_SUCCESS) {
-            return $value;
+        if ($ttl === null) {
+            $expiration = static::EXPIRATION_INFINITY;
+        } elseif ($ttl <= 0) {
+            $expiration = static::EXPIRATION_EXPIRED;
+        } else {
+            $expiration = $ttl + time();
         }
 
-        return $default;
+        return $expiration;
     }
 
-    protected function getValues(iterable $keys, $default = null): iterable
+    /**
+     * Normalizes cache TTL handling `null` value and {@see DateInterval} objects.
+     * @param int|DateInterval|null $ttl raw TTL.
+     * @return int|null TTL value as UNIX timestamp or null meaning infinity
+     */
+    private function normalizeTtl($ttl): ?int
     {
-        $values = $this->cache->getMulti($keys);
-
-        if ($this->cache->getResultCode() === \Memcached::RES_SUCCESS) {
-            return $values;
+        if ($ttl instanceof DateInterval) {
+            try {
+                return (new DateTime('@0'))->add($ttl)->getTimestamp();
```

Comment:
When would it throw?

Reply:
I think never, as we pass constant to `DateTime`, but it's declared in the `DateTime` doc block that it throws an exception and if don't catch it in the `normalizeTtl` we will have to specify `@throws Exception` tag also.

Reply:
> Do you know a better way to deal with this?

What about doing nothing? This is only false-positive from inspection.

Reply:
I'd just remove it and let it complain or use special IDE annotations such as `/** @noinspection PhpDocMissingThrowsInspection */` to make it silent.